flink-cdc-base/src/main/java/com/ververica/cdc/connectors/base/utils/SourceRecordUtils.java

```diff
@@ -139,8 +140,17 @@ public static TableId getTableId(SourceRecord dataRecord) {
         Struct value = (Struct) dataRecord.value();
         Struct source = value.getStruct(Envelope.FieldName.SOURCE);
         String dbName = source.getString(DATABASE_NAME_KEY);
+        // Oracle need schemaName
+        String schemaName = getSchemaName(source);
         String tableName = source.getString(TABLE_NAME_KEY);
-        return new TableId(dbName, null, tableName);
+        return new TableId(dbName, schemaName, tableName);
+    }
+
+    public static String getSchemaName(Struct source) {
+        if (source.schema().fields().stream().anyMatch(r -> SCHEMA_NAME_KEY.equals(r.name()))) {
+            return source.getString(SCHEMA_NAME_KEY);
+        }
+        return null;
```

Comment:
It is better to add a test for this method

Reply:
I also think we only need changes in this class and add some tests for it. Other changes need to be removed from this PR.